File: java/src/it/java/io/github/adikeshri/tachyon/TestSupport.java
package io.github.adikeshri.tachyon;

import java.util.UUID;
import java.util.function.Consumer;

final class TestSupport {
    private TestSupport() {
    }

    static String baseUrl() {
        String value = System.getenv("TACHYON_URL");
        return value == null || value.isEmpty() ? "http://localhost:8108" : value;
    }

    static String adminKey() {
        String value = System.getenv("TACHYON_ADMIN_KEY");
        return value == null || value.isEmpty() ? "admin-key" : value;
    }

    static String searchKey() {
        String value = System.getenv("TACHYON_SEARCH_KEY");
        return value == null || value.isEmpty() ? "search-key" : value;
    }

    static TachyonClient adminClient() {
        return TachyonClient.builder(baseUrl()).apiKey(adminKey()).build();
    }

    static TachyonClient searchOnlyClient() {
        return TachyonClient.builder(baseUrl()).apiKey(searchKey()).build();
    }

    static TachyonClient anonymousClient() {
        return TachyonClient.builder(baseUrl()).build();
    }

    static String uniqueName(String prefix) {
        return prefix + "-" + UUID.randomUUID();
    }

    /**
     * Creates a collection, runs {@code body}, and deletes the collection
     * afterwards — even if {@code body} throws — so integration tests never
     * leak collections into the shared server.
     */
    static void withCollection(TachyonClient client, CollectionSchema schema, Consumer<Collection> body) {
        client.collections.create(schema);
        try {
            body.accept(client.collection(schema.name()));
        } finally {
            try {
                client.collections.delete(schema.name());
            } catch (RuntimeException e) {
                // best-effort cleanup
            }
        }
    }
}

File: csharp/Tachyon.Sdk.IntegrationTests/TestSupport.cs
namespace Tachyon.Sdk.IntegrationTests;

public static class TestSupport
{
    public static string BaseUrl => Environment.GetEnvironmentVariable("TACHYON_URL") ?? "http://localhost:8108";
    public static string AdminKey => Environment.GetEnvironmentVariable("TACHYON_ADMIN_KEY") ?? "admin-key";
    public static string SearchKey => Environment.GetEnvironmentVariable("TACHYON_SEARCH_KEY") ?? "search-key";

    public static TachyonClient AdminClient() => new(new TachyonClientOptions { Url = BaseUrl, ApiKey = AdminKey });
    public static TachyonClient SearchOnlyClient() => new(new TachyonClientOptions { Url = BaseUrl, ApiKey = SearchKey });
    public static TachyonClient AnonymousClient() => new(new TachyonClientOptions { Url = BaseUrl });

    public static string UniqueName(string prefix) => $"{prefix}-{Guid.NewGuid():N}";

    /// <summary>
    /// Creates a collection, runs <paramref name="body"/>, and deletes the
    /// collection afterwards — even if <paramref name="body"/> throws — so
    /// integration tests never leak collections into the shared server.
    /// </summary>
    public static async Task WithCollectionAsync(TachyonClient client, CollectionSchema schema, Func<Collection, Task> body)
    {
        await client.Collections.CreateAsync(schema);
        try
        {
            await body(client.Collection(schema.Name));
        }
        finally
        {
            try
            {
                await client.Collections.DeleteAsync(schema.Name);
            }
            catch
            {
                // best-effort cleanup
            }
        }
    }
}
